utils/proxy/core.py

```diff
@@ -81,10 +82,16 @@ def _scrub(self, content):
         return content
 
     def request(self, flow):
-        if flow.request.host in ("runner", "localhost"):  # localhost because on UDS mode, UDS socket is redirected
-            flow.request.host, flow.request.port = "agent", 8126
+        # localhost because on UDS mode, UDS socket is redirected
+        if flow.request.host in ("runner", "localhost", "host.docker.internal", "host-gateway"):
+            flow.request.host, flow.request.port = "localhost", 8127
             flow.request.scheme = "http"
 
+    @staticmethod
+    def request_is_from_tracer(request):
+        # as now, only the tracer use the proxy as a reverse proxy
```

Comment:
Not accurate since the E2E tests use the reverse proxy mode for the tests making requests to the backend. Unless you mean on port `8127` specifically.